### PR TITLE
Feature/rarimo checkout

### PR DESCRIPTION
**Pull request by SegmentationFaultEnjoyer**
Integration with Rarimo checkout 

[https://rarimo.github.io/js-sdk/modules/_rarimo_nft_checkout.html](url)

### CODE REVIEW

File: src/composables/use-nft-checkout.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    const providerInstance = await createProvider(MetamaskProvider)
+
+    checkout = createCheckoutOperation(EVMOperation, providerInstance)
+  }
+
+  const getEstimatedPrice = async (paymentToken: PaymentToken) => {
+    return checkout && checkout.estimatePrice(paymentToken)
+  }
+
+  const sendTransaction = async (
```
Comment (lukachi) [inline src/composables/use-nft-checkout.ts:118]: bad naming

File: src/composables/use-nft-checkout.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    const bundle = utils.defaultAbiCoder.encode(
+      ['address[]', 'uint256[]', 'bytes[]'],
+      [[marketPlaceAddress], [txOpts.amountOfEth], [encodedFunctionData]],
+    )
+
+    const txHash = await checkout.checkout(estimatedPrice, { bundle })
+
+    return txHash
+  }
```
Comment (lukachi) [inline src/composables/use-nft-checkout.ts:145]: do not merge contract interaction with local hooks

File: src/forms/PurchaseBookForm.vue [hunk 3]
```diff
@@ -215,89 +266,83 @@ const mintToken = async (
...
+      },
+    },
+    signature: {
+      ...mintSignature.signature,
+      endSigTimestamp: mintSignature.end_timestamp,
+    },
+    nativeTokenAmount,
+  }
+
+  if (paymentType.value === PAYMENT_TYPES.rarimo) {
```
Comment (lukachi) [inline src/forms/PurchaseBookForm.vue:313]: bad idea to set conditions for specific chain

File: src/forms/purchase-book-payments/RarimoTemplate.vue [hunk 4]
```diff
@@ -0,0 +1,317 @@
+<template>
+  <template v-if="!isLoading">
```
Comment (lukachi) [inline src/forms/purchase-book-payments/RarimoTemplate.vue:2]: true first

File: src/pages/bookshelf/BookshelfItemPage.vue [hunk 5]
```diff
@@ -46,11 +46,17 @@
 
           <book-details :book="book" />
 
+          <bookshelf-rarimo-message
```
Comment (lukachi) [inline src/pages/bookshelf/BookshelfItemPage.vue:49]: rarimo in abstract component

File: src/composables/use-nft-checkout.ts (hunk 1 above)
Comment (SegmentationFaultEnjoyer) [reply src/composables/use-nft-checkout.ts:118]: fixed

File: src/forms/purchase-book-payments/RarimoTemplate.vue (hunk 4 above)
Comment (SegmentationFaultEnjoyer) [reply src/forms/purchase-book-payments/RarimoTemplate.vue:2]: fixed

File: src/pages/bookshelf/BookshelfItemPage.vue (hunk 5 above)
Comment (SegmentationFaultEnjoyer) [reply src/pages/bookshelf/BookshelfItemPage.vue:49]: renamed to more appropriate name

**Pull request merged**